### PR TITLE
Require at least node >=14.13.0

### PR DESCRIPTION
**Pull request by CraigMacomber**
## Description

Enforce that the version of node used is at least 14.13.0 (released on 2020-09-29), which according to https://www.the-guild.dev/blog/support-nodejs-esm means it should support ESM. This uses the mechanism documented in https://docs.npmjs.com/cli/v8/configuring-npm/package-json#engines.

This produces an error when running `npm i` in the root if using an older version of node.

Note that the 14 branch of node is the oldest supported branch (see https://nodejs.org/en/about/releases/), no one should be one node older than 14, so updating to at least 14.13.0 should be a non-breaking change.

Additionally, our `.nvmrc` file specifies node 14, and our readme says to use that version.

 Since this is in the root package.json, not one we ship to users, it should have no impact on consumers of our packages.

## Reviewer Guidance

I'm not aware of us placing any restrictions on node versions within the 14 range in the past.
This change is intended to introduce a new approach for being explicit about how old of Node we support for development.

I'd like opinions on if this is a good approach to take, and if this is a good version to start with.
I'm not aware of any other specific features that were added to 14 after release that we might care about: if you have a reason to pick some other version let me know.

## Example Error

If I force requiring version 15 instead (so my system errors) the error looks like:

```
$ npm i
npm ERR! code ENOTSUP
npm ERR! notsup Unsupported engine for root@0.14.0: wanted: {"node":">=15.13.0"} (current: {"node":"14.19.3","npm":"6.14.17"})
npm ERR! notsup Not compatible with your version of node/npm: root@0.14.0
npm ERR! notsup Not compatible with your version of node/npm: root@0.14.0
npm ERR! notsup Required: {"node":">=15.13.0"}
npm ERR! notsup Actual:   {"npm":"6.14.17","node":"14.19.3"}
```

I have confirmed `npm ci` also produces this error.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -122,6 +122,9 @@
     "run-script-os": "^1.1.6",
     "typescript": "~4.5.5"
   },
+  "engines": {
+    "node": ">=15.13.0"
```
Comment (Abe27342) [inline package.json:126]: leftover change from your description :-)

File: package.json [hunk 2]
```diff
@@ -122,6 +122,9 @@
     "run-script-os": "^1.1.6",
     "typescript": "~4.5.5"
   },
+  "engines": {
```
Comment (curtisman) [inline package.json:125]: Should we do this to all the package.json?

File: .npmrc [hunk 3]
```diff
@@ -0,0 +1 @@
+engine-strict=true
```
Comment (Abe27342) [inline .npmrc:1]: we should update the two `.nvmrc` files in the repo to 14.13 as well

File: package.json (hunk 1 above)
Comment (curtisman) [inline package.json:126]: 15.13.0?  the title says 14.13.0.

Comment (CraigMacomber) [reply package.json]: Oops, I checked the version I used for testing the errors when fixing the package sorting! Fixed.

File: package.json (hunk 2 above)
Comment (CraigMacomber) [reply package.json:125]: If we want to force users of our packages to use specific versions, maybe as a future change. For now though I just want to ensure none of our devs (or CI) is on an older version, which this should accomplish in practice.

File: package.json (hunk 1 above)
Comment (CraigMacomber) [reply package.json]: Fixed.

File: .npmrc (hunk 3 above)
Comment (CraigMacomber) [reply .npmrc:1]: Is there a string that you think would be better for nvm than "14"?

As far as I can tell (And I don't use nvm), there isn't a way to say >=14.13.0 in a way nvm understands. Some discussion here https://github.com/nvm-sh/nvm/issues/2562 . I don't want to force an old version of the 14 branch on nvm users, and 14.13.* is pretty old at this point.

**Comment by CraigMacomber** [comment]
Build is failing due to modified .npmrc:

```
	modified:   .npmrc
##[error]Build should not create extraenous files

```

I'm not sure what to do about that. Is there any way to see what modifications it found?

File: package.json (hunk 2 above)
Comment (curtisman) [reply package.json:125]: How about the other mono repos (server and azure)

**Comment by tylerbutler** [comment]
> I'm not sure what to do about that. Is there any way to see what modifications it found?

I don't think there's an easy way to see the file content unless it wound up in a build artifact. Ideally the build task that checks for modified files would output the necessary stuff to debug, but it doesn't. Code is here:

https://github.com/microsoft/FluidFramework/blob/fcdbc91381d3ffe179a611a7f55ed418018be4e2/tools/pipelines/templates/build-npm-package.yml#L382-L392

Quick and dirty fix might be to exclude .npmrc.

My guess is that the CI pipeline itself is responsible for the changes. We add/update `.npmrc` in order to publish to private feeds:

https://github.com/microsoft/FluidFramework/blob/fcdbc91381d3ffe179a611a7f55ed418018be4e2/tools/pipelines/templates/include-publish-npm-package-steps.yml#L28-L62

However, that step is only after we pack, and seems to be in a subfolder so it _should_ be isolated. I didn't look at the code very deeply, though.

File: .npmrc (hunk 3 above)
Comment (Abe27342) [reply .npmrc:1]: Yeah, that's unfortunate. I never actually looked into these files, just was aware of their existence. Leaving them both at `14` is probably best then.

**Pull request closed without merging**